cdapython/results/base.py

```diff
@@ -246,6 +249,15 @@ async def tmp() -> AsyncGenerator[Any, None]:
 
         return tmp()
 
+    def sort(self, function):
+        # -1 , 1 , 0
+        self._result = sorted(self._result, key=function)
+        return copy(self)
```

Comment:
this may be a misunderstanding about python. but my guess is that you are trying to leave the self._result unsorted but return an new copy of self that has a sorted result. It looks like you are changing the result on the referenced object and then returning a copy of it. I think either you should copy and then sort the result in the copy and return it or not do the copy at all.

Reply:
thanks that is a good catch i missed this.